src/frontend/screens/Onboarding/AddToProjectScreen/index.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from "react";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { View, StyleSheet } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+
+import ConfigContext from "../../../context/ConfigContext";
+import { MEDIUM_BLUE, WHITE } from "../../../lib/styles";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { BackIcon } from "../../../sharedComponents/icons";
+import IconButton from "../../../sharedComponents/IconButton";
+import { WithWifiBar } from "../WithWifiBar";
+
+import { ScanQrCodeStep } from "./ScanQrCodeStep";
+import { DeviceFoundStep } from "./DeviceFoundStep";
+import { SuccessStep } from "./SuccessStep";
+
+type Step = "scan" | "found" | "success";
+
+const m = defineMessages({
+  titleGeneric: {
+    id: "screens.Onboarding.AddToProjectScreen.titleGeneric",
+    defaultMessage: "Add to Project",
+  },
+});
+
+export const AddToProjectScreen: NavigationStackScreenComponent = () => {
+  const [step, setStep] = React.useState<Step>("scan");
+  const [foundDeviceId, setFoundDeviceId] = React.useState<string>();
+  const [config] = React.useContext(ConfigContext);
+
+  const getRenderedStep = () => {
+    switch (step) {
+      case "scan":
+        return (
+          <ScanQrCodeStep
+            goNext={data => {
+              setFoundDeviceId(data);
+              setStep("found");
+            }}
+          />
+        );
+      case "found":
+        return (
+          !!foundDeviceId && (
+            <DeviceFoundStep
+              deviceId={foundDeviceId}
+              goNext={() => setStep("success")}
+              goBack={() => {
+                setFoundDeviceId(undefined);
+                setStep("scan");
+              }}
+            />
+          )
+        );
+      case "success":
+        return (
+          !!foundDeviceId &&
+          !!config.metadata.name && (
+            <SuccessStep
+              goNext={() => {
+                setFoundDeviceId(undefined);
+                // TODO: Need to go to a sync screen instead
+                setStep("scan");
+              }}
+              deviceId={foundDeviceId}
+              projectName={config.metadata.name}
+            />
+          )
+        );
+    }
+  };
+
+  return (
+    <WithWifiBar>
+      <View style={styles.container}>{getRenderedStep()}</View>
+    </WithWifiBar>
+  );
+};
+
+AddToProjectScreen.navigationOptions = () => ({
+  animationEnabled: false,
```

Comment:
had to disable animation here because the camera component makes the entering screen transition incredibly choppy. open to ideas on how to fix

Reply:
fixed this using https://github.com/react-navigation/hooks#usenavigationeventshandler